ckeditor/static/ckeditor/ckeditor/plugins/divepullquote/dialogs/divepullquote.js

```diff
@@ -0,0 +1,88 @@
+/**
+Pullquote plugin - adds icon to editor that opens a dialog with 4 text inputs that allow a user to create a pull quote.
+This dialog shows the user 4 text inputs, and allows them to input a quote, image url, speaker name, and speaker-title
+
+- This plugin is a widget (made using the widget plugin), so this dialog edits the widget's data property that
+corresponds to the pullquote element that the user is editing in the dialog
+
+*/
+CKEDITOR.dialog.add('divepullquote', function(editor){
+
+	return {
+		title: 'Insert pull quote',
+		minWidth: 400,
+		minHeight: 200,
+		contents: [
+			{
+				id: 'divepullquotedialog',
+				elements: [
+					{
+						type: 'text',
+						id: 'pq-quote',
+						label: 'Quote',
+						validate: CKEDITOR.dialog.validate.notEmpty( "Quote field cannot be empty." ),
+						setup: function( widget ) {
+							this.setValue(widget.data.quote_value);
+						},
+						commit: function( widget ) {
+							widget.setData('quote_value', this.getValue());
+						}
+					},
+					{
+						//image url input -- dont validate for empty input...see commit function below
+						type: 'text',
+						id: 'pq-headshot-img',
+						label: 'Pullquote Image URL (ex. http://www.educationdive.com/user_media/diveimage/johndoe.jpg )',
+						//validate: CKEDITOR.dialog.validate.notEmpty( "Image source field cannot be empty." ),
+						setup: function( widget ) {
+							this.setValue(widget.data.img_src);
+						},
+						commit: function( widget ) {
+							//get the actual img dom element, not some ckeditor psuedo element wrapped bullshit
+							var img = widget.editables.imgDiv.getFirst().$;
+
+							//checks for an empty value or one space...we allow user to create pullquote without an image,
+							//but we need to change the pullquote class styling to not display on the page
+							if(this.getValue() === '' || this.getValue() === ' '){
+								widget.setData('img_src', this.getValue());
```

Comment:
Can we not just remove the img node here if it isn't being used? Something like `img.parentNode.removeChild(img)`